scripts/bfasst/compare/waveform.py

```diff
@@ -0,0 +1,355 @@
+        TCL.write("gtkwave::File/Quit")
+
+    def generateTCL(self, PATH, fileName, fileNum):
+        if(exists(PATH + fileName[fileNum] + ".tcl")):
+            os.remove(PATH + fileName[fileNum] + ".tcl")   
+        TCL = open(PATH + fileName[fileNum] + ".tcl", "x")
+        sample = open(PATH + fileName[fileNum-1] + ".tcl")
+        for line in sample:
+            if fileName[fileNum-1] in line:
+                line = line.replace(fileName[fileNum-1], fileName[fileNum])
+            TCL.write(line)
+        
+
+    def generate_files(self, design):
+        TESTS = 100 #Note: change this number to however many tests the testbench should run.
+        SAMPLE_PATH = str(pathlib.Path.cwd()) + "/scripts/bfasst/compare_waveforms/sample_tb.v"
+        fileName = [design.impl_netlist_path.name[0: len(design.impl_netlist_path.name)-2], 
+        design.reversed_netlist_path.name[0: len(design.reversed_netlist_path.name)-2]]
+        file = [open(design.impl_netlist_path), open(design.reversed_netlist_path)]
+        fileNum = 0
+        PATH = str(design.impl_netlist_path)
+        PATH = PATH[0: len(PATH) - len(design.impl_netlist_path.name)]
+
+        for x in file:
+
+            if(x.name.find("reversed") != -1):
+                self.fix_file(PATH, fileName[fileNum], True)
+                self.parseReversed(PATH, x.name)
+            else:
+                self.fix_file(PATH, fileName[fileNum], False)
+                self.parse(x.name)
+            
+            if(exists(PATH + fileName[fileNum] + "_tb.v")):
+                os.remove(PATH + fileName[fileNum] + "_tb.v")
+            
+            if fileNum == 0:
+                sample = open(SAMPLE_PATH)
+            else:
+                sample = open(PATH + fileName[fileNum-1] + "_tb.v")
+            tb = open(PATH + fileName[fileNum] + "_tb.v", "x")
+
+            for line in sample:
+                if(fileNum == 0):
+                    self.generateFirstTestbench(tb, line, TESTS, fileName)
+                else:
+                    self.generateTestbench(tb, line, fileName, fileNum)
+            
+            sample.close()
+            tb.close()
+
+            if (fileNum == 0):
+                self.generateFirstTCL(PATH, fileName, fileNum)
+            else:
+                self.generateTCL(PATH, fileName, fileNum)
+
+            fileNum = fileNum + 1
+            refresh(data)
+
+    def runTest(self, design):
+        isEquivalent = False
+        IMPL_NAME = design.impl_netlist_path.name[0: len(design.impl_netlist_path.name)-2]
+        PATH = str(design.impl_netlist_path)
+        PATH = PATH[0: len(PATH) - len(design.impl_netlist_path.name)]
+        string = "iverilog -o " + PATH + "dsn " + PATH + IMPL_NAME + "_tb.v " 
+        string = string + PATH + design.impl_netlist_path.name + " " + str(pathlib.Path.cwd()) + "/third_party/yosys/techlibs/xilinx/cells_sim.v"
```

Comment:
Do we have something in `paths` for this already?

Reply:
I added to paths to have the build folder and the xilinx_yosys_waveform build folder inside of it. Now, I can call 
`bfasst.paths.WAVEFORM_BUILD / design.rel_path` to achieve the same result as `PATH = str(design.impl_netlist_path)
        PATH = PATH[0: len(PATH) - len(design.impl_netlist_path.name)]`